packages/webview_flutter/webview_flutter_wkwebview/lib/src/webview_cupertino_widget.dart

```diff
@@ -0,0 +1,227 @@
+  static final Future<IosDeviceInfo> _defaultDeviceInfo =
+      DeviceInfoPlugin().iosInfo;
+
+  static Future<String> get _defaultSystemVersion async {
+    final String? systemVersion = (await _defaultDeviceInfo).systemVersion;
+    assert(
+      systemVersion != null,
+      'Unable to retrieve the system version of this device.',
+    );
+    return systemVersion!;
+  }
+
+  final Completer<void> _initializationCompleter = Completer<void>();
+
+  /// Handles callbacks that are made by navigation.
+  final WebViewPlatformCallbacksHandler callbacksHandler;
+
+  /// Manages named JavaScript channels and forwarding incoming messages on the correct channel.
+  final JavascriptChannelRegistry javascriptChannelRegistry;
+
+  /// Handles constructing a [web_kit.WebView].
+  ///
+  /// This should only be changed when used for testing.
+  final WebViewProxy webViewProxy;
+
+  /// Represents the WebView maintained by platform code.
+  late final web_kit.WebView webView;
+
+  /// The system version of the iOS device.
+  late final FutureOr<String> systemVersion;
+
+  Future<void> _initialize() => _initializationCompleter.future;
+
+  Future<void> _setCreationParams(
+    CreationParams params, {
+    required web_kit.WebViewConfiguration configuration,
+  }) async {
+    await _setWebViewConfiguration(
+      configuration,
+      allowsInlineMediaPlayback: params.webSettings?.allowsInlineMediaPlayback,
+      autoMediaPlaybackPolicy: params.autoMediaPlaybackPolicy,
+    );
+
+    webView = webViewProxy.createWebView(configuration);
+  }
+
+  Future<void> _setWebViewConfiguration(
+    web_kit.WebViewConfiguration configuration, {
+    required bool? allowsInlineMediaPlayback,
+    required AutoMediaPlaybackPolicy autoMediaPlaybackPolicy,
+  }) async {
+    if (allowsInlineMediaPlayback != null) {
+      configuration.allowsInlineMediaPlayback = allowsInlineMediaPlayback;
+    }
+
+    late final bool requiresUserAction;
+    switch (autoMediaPlaybackPolicy) {
+      case AutoMediaPlaybackPolicy.require_user_action_for_all_media_types:
+        requiresUserAction = true;
+        break;
+      case AutoMediaPlaybackPolicy.always_allow:
+        requiresUserAction = false;
+        break;
+    }
+
+    if (_isGreaterOrEqualVersion(await systemVersion, '10.0')) {
+      configuration.mediaTypesRequiringUserActionForPlayback =
+          <web_kit.AudiovisualMediaType>{
+        if (requiresUserAction) web_kit.AudiovisualMediaType.all,
+        if (!requiresUserAction) web_kit.AudiovisualMediaType.none
+      };
+    } else if (_isGreaterOrEqualVersion(await systemVersion, '9.0')) {
+      configuration.requiresUserActionForMediaPlayback = requiresUserAction;
+    } else {
+      configuration.mediaPlaybackRequiresUserAction = requiresUserAction;
+    }
+  }
+
+  bool _isGreaterOrEqualVersion(String first, String second) {
+    return _parseVersion(first).compareTo(_parseVersion(second)) >= 0;
+  }
+
+  Version _parseVersion(String versionStr) {
```

Comment:
`pub_semver` is overkill here; you're already writing manual parsing code anyway, and the set of things it needs to support is smaller than `pub_semver` (there's no `iOS 10.0.7+3-dev` for instance). You can eliminate the dependency and just parse to a three-element array then compare in a loop.